Sources/ClientRuntime/Networking/Http/SdkHttpRequest.swift

```diff
@@ -49,6 +49,17 @@ extension SdkHttpRequest {
         httpRequest.body = StreamableHttpBody(body: body)
         return httpRequest
     }
+
+    public func toHttp2Request() throws -> HTTPRequestBase {
```

Comment:
Why do we return a `HTTPRequestBase` here but the `toHttpRequest()` returns a `HTTPRequest`?

Can you add comments describing what makes this a http2 request? It looks almost identical to `toHttpRequest` except that body is nil. Why does setting `body` to nil make this a http2 request?

Reply:
added comment for it